### PR TITLE
Contest

### PR DESCRIPTION
**Pull request by Changron**

### CODE REVIEW

File: nthuoj/urls.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from django.contrib import admin
 
 urlpatterns = patterns('',
-    url(r'^admin/', include(admin.site.urls)),
```
Comment (nickelcheng) [inline nthuoj/urls.py]: yes, please keep it. It's a link to backend(database) management page.

File: contest/templates/contest/contestArchive.html [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                  {% endif %}
+                </tbody>
+              </table>
+            </div>
+            <div class="tab-pane" id="contestant">
+              <table class="table" style="text-align:center">
+                <thead>
+                  <tr>
+                    {% if admin %}
+                    <th></th>
```
Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: I mean your \<th\>\</th\> is in if statement, maybe you should add two more spaces in front of it. (so does other {% if ... %} or {% for ... %} tags)

Comment (Changron) [inline contest/templates/contest/contestArchive.html]: In Henry's profile.html line 79, there's no indent too.
On the other hand, if use Sublime prettify plugin, it's what it looks like after using it.
In django's offical tutorial, {%%}  tag's indent follows it's parent's.
That is,we should write something like this,
<tr>
{% if admin %}
  <th></th>

Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: OK, I'm sorry. Please ignore the comment.

Comment (Changron) [inline contest/templates/contest/contestArchive.html]: umm
actually i mean, in django, they have same indent as < tr >
but both henry and me have same indent as < th >
should we follow them or use our own style?

File: contest/templates/contest/contestArchive.html [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+                    <th style="text-align:center">#</th>
+                    <th style="text-align:center">Problem</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  {% for problem in contest.problem_list %}
+                  <tr>
+                    {% if admin %}
+                    <td>
+                      <button type="button" class="btn btn-warning btn-xs"> <span class="glyphicon glyphicon-refresh"></span>
```
Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: Do not write over 80 characters in one line.

File: contest/templates/contest/contestArchive.html [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+                      </button>
+                      <button type="button" class="btn btn-danger btn-xs"> <span class="glyphicon glyphicon-minus"></span>
```
Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: Do not write over 80 characters in one line.

File: contest/templates/contest/contestArchive.html [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+                    {% endif %}
+                    <td>{{ problem.id }}</td>
+                    <td>{{ problem.name }}</td>
+                  </tr>
+                  {% endfor %}
+                  <!-- -->
+                  {% if admin %}
+                  <tr>
+                    <td style="width:10%">
+                      <button type="button" class="btn btn-success btn-xs"> <span class="glyphicon glyphicon-plus"></span>
```
Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: Do not write over 80 characters in one line.

File: contest/templates/contest/contestArchive.html [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+                    <th style="text-align:center">#</th>
+                    <th style="text-align:center">Contestant</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  {% for ppl in contest.contestant_list %}
+                  <tr>
+                    {% if admin %}
+                    <td>
+                      <button type="button" class="btn btn-danger btn-xs"> <span class="glyphicon glyphicon-minus"></span>
```
Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: Do not write over 80 characters in one line.

File: contest/templates/contest/contestArchive.html (hunk 2 above)
Comment (nickelcheng) [inline contest/templates/contest/contestArchive.html]: We can use our own style.

**Comment by nickelcheng** [comment]
I'm OK to this PR.

**Comment by bruce3557** [comment]
:+1: , going to merge

**Pull request merged**